src/metabase/automagic_dashboards/core.clj

```diff
@@ -660,92 +781,80 @@
 
 (defmethod automagic-analysis (type Table)
   [table opts]
-  (automagic-dashboard (merge opts (->root table))))
+  (automagic-dashboard (merge (->root table) opts)))
 
 (defmethod automagic-analysis (type Segment)
   [segment opts]
-  (automagic-dashboard (merge opts (->root segment))))
+  (automagic-dashboard (merge (->root segment) opts)))
 
 (defmethod automagic-analysis (type Metric)
   [metric opts]
-  (automagic-dashboard (merge opts (->root metric))))
-
-(def ^:private ^{:arglists '([x])} encode-base64-json
-  (comp codec/base64-encode codecs/str->bytes json/encode))
-
-(def ^:private ^{:arglists '([card-or-question])} nested-query?
-  (comp (every-pred string? #(str/starts-with? % "card__"))
-        #(qp.util/get-in-normalized % [:dataset_query :query :source_table])))
-
-(def ^:private ^{:arglists '([card-or-question])} native-query?
-  (comp #{:native} qp.util/normalize-token #(qp.util/get-in-normalized % [:dataset_query :type])))
-
-(def ^:private ^{:arglists '([card-or-question])} source-question
-  (comp Card #(Integer/parseInt %) second #(str/split % #"__")
-        #(qp.util/get-in-normalized % [:dataset_query :query :source_table])))
+  (automagic-dashboard (merge (->root metric) opts)))
+
+(defn- decompose-question
```

Comment:
Seems like a good candidate for being split, it's got three levels of nested `->>` and `->`